src/core/xref-db.js

```diff
@@ -0,0 +1,79 @@
+// @ts-check
+import { IDBKeyVal } from "./utils.js";
+
+/**
+ * @typedef {import('core/xref').RequestEntry} RequestEntry
+ * @typedef {import('core/xref').Response} Response
+ * @typedef {import('core/xref').SearchResultEntry} SearchResultEntry
+ */
+
+const CACHE_MAX_AGE = 86400000; // 24 hours
+
+/**
+ * Temporary workaround until browsers get import-maps
+ */
+async function importIdb() {
+  try {
+    return await import("idb");
+  } catch {
+    return await import("../../node_modules/idb/build/esm/index.js");
+  }
+}
+
+async function getIDBKeyValForXref() {
```

Comment:
Nit: `getXrefCache`?

Reply:
Bit confusing whether it's for retrieving cache data or an IDB handle...

Reply:
As we have `const cache = await getIDBKeyValForXref();`, so I'll say it gets the cache.